core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -361,9 +362,13 @@ private int getUnixFileMode(final String pathAsString) {
         if (this.forcedFileMode != null) {
             return this.getModeValue(path);
         }
+        return getUnixFileMode(path);
+    }
 
+    @UnstableAPI
+    public static int getUnixFileMode(final Path path) {
         try {
-            return (int) Files.getAttribute(path, "unix:mode");
+            return (int) Files.readAttributes(path, "unix:mode").get("mode");
```

Comment:
I was curious as to why we'd need to use `readAttributes` here - we're still only fetching the mode, so `getAttribute` perhaps doesn't really have a performance disadvantage.

_However_, now this makes me wonder if we should also be hashing [basic file attributes](https://docs.oracle.com/javase/8/docs/api/java/nio/file/attribute/BasicFileAttributeView.html) like created/modified timestamps etc 😭 

WDYT? I don't mind deferring this, TBH...

Reply:
since we checksum content & file mode (the only bits that actually go into the tar archive), I guess we can defer the basic file attributes until someone reports that the hashing works incorrectly for them.

I will also check what Dockerfile builder is using for hashing `COPY`

Reply:
Cool, sounds sensible :+1:

Reply:
https://stackoverflow.com/a/59073724/1826422

> For the ADD and COPY instructions, the contents of the file(s) in the image are examined and a checksum is calculated for each file. The last-modified and last-accessed times of the file(s) are not considered in these checksums. During the cache lookup, the checksum is compared against the checksum in the existing images. If anything has changed in the file(s), such as the contents and metadata, then the cache is invalidated.

okay, looks pretty aligned with what we're doing